### PR TITLE
Integration testing

### PR DESCRIPTION
**Pull request by rndmh3ro**
* add script to update test and example playbooks
* add integration tests for local and github testing
* add local testing section to readme
* update examples and test playbook

### CODE REVIEW

File: hacking/update_examples_and_tests.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 for module in ../plugins/modules/*.py; do
```
Comment (schurzi) [inline hacking/update_examples_and_tests.sh:3]: directory `hacking/`?

File: tests/integration/test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        timeout: 60
+
+    - include_tasks: "{{ item }}"
+      loop:
+        - ./icinga_command_template.yml
+        - ./icinga_command.yml
+        - ./icinga_timeperiod.yml
+        - ./icinga_user_template.yml
+        - ./icinga_user.yml
+        # - ./icinga_notification.yml
```
Comment (schurzi) [inline tests/integration/test.yml:23]: some are commented out. Are here problems with these objects?

File: tests/integration/icinga_notification.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+- name: create notification
+  icinga_notification:
+    state: present
+    url: "{{ icinga_url }}"
+    url_username: "{{ icinga_user }}"
+    url_password: "{{ icinga_pass }}"
+    apply_to: host
+    assign_filter: '"E-Mail"=host.vars.enabled_notifications'
```
Comment (schurzi) [inline tests/integration/icinga_notification.yml:9]: could we swap the leements in the comparison? This order triggers my OCD

File: hacking/update_examples_and_tests.sh (hunk 1 above)
Comment (rndmh3ro) [reply hacking/update_examples_and_tests.sh:3]: Yes. :) I need to put it somewhere.

File: tests/integration/icinga_notification.yml (hunk 3 above)
Comment (rndmh3ro) [reply tests/integration/icinga_notification.yml:9]: Sure. This way copy-pasted from production-code. :)

File: hacking/update_examples_and_tests.sh (hunk 1 above)
Comment (schurzi) [reply hacking/update_examples_and_tests.sh:3]: very well, I usually expect a directory `bin/` or `scripts/` for that ^^

File: tests/integration/test.yml (hunk 2 above)
Comment (rndmh3ro) [reply tests/integration/test.yml:24]: Fixed most of them. For notifications there's a notification template module needed.

File: .yamllint [hunk 4]
```diff
@@ -13,3 +13,4 @@ rules:
   brackets:
     max-spaces-inside: 1
     level: error
+  line-length: disable
```
Comment (schurzi) [inline .yamllint:16]: is that one needed now, or is it an artefact from an earlier commit?

Comment (rndmh3ro) [reply .yamllint]: Fixed. Was an artefact.

**Pull request merged**